pandas/core/ops.py

```diff
@@ -618,14 +619,15 @@ def f(x):
 
         return lvalues, rvalues
 
-    def _is_offset(self, arr_or_obj):
-        """ check if obj or all elements of list-like is DateOffset """
-        if isinstance(arr_or_obj, ABCDateOffset):
-            return True
-        elif (is_list_like(arr_or_obj) and len(arr_or_obj) and
-              is_object_dtype(arr_or_obj)):
-            return all(isinstance(x, ABCDateOffset) for x in arr_or_obj)
-        return False
+
+def _is_offset(arr_or_obj):
```

Comment:
better to move this to pandas.core.dtypes.common and de-privatize

Reply:
and add tests

Reply:
Agreed on all three points.